app/main.py

```diff
@@ -1,2 +1,70 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    INITIAL_CAPACITY = 8
+    LOAD_FACTOR = 2 / 3
+
+    def __init__(self) -> None:
+        self.capacity = self.INITIAL_CAPACITY
+        self.length = 0
+        self.hash_table = [[] for _ in range(self.INITIAL_CAPACITY)]
+
+    def hash_table_index(self, key: Hashable) -> int:
+        hashed_key = hash(key)
+        index = hashed_key % self.capacity
+
+        return index
+
+    def __len__(self) -> int:
+        return self.length
+
+    def resize(self) -> None:
+        threshold = self.capacity * self.LOAD_FACTOR
+
+        if self.length > threshold:
+            self.capacity *= 2
+            old_hash_table = self.hash_table
+
+            self.hash_table = [[] for _ in range(self.capacity)]
+
+            for item in old_hash_table:
+                if item:
+                    key, value = item
+                    self.__setitem__(key, value)
+                    self.length -= 1
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self.hash_table_index(key)
+
+        while self.hash_table[index]:
+            key_, value_ = self.hash_table[index]
+            if key == key_:
+                self.hash_table[index] = (key, value)
```

Comment:
sorry) you need to add hash as well:
`Also, not forgot to store (key, hash, value) as node in hash table.`